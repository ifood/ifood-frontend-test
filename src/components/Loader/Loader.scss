@import '../../assets/styles/variables';

.loader {
  position: relative;
  width: 100%;
  height: 150px;

  > .move {
    position: absolute;
    top: 50%;
    left: 50%;
    width: 3rem;
    height: 3rem;
    animation: spin 3.5s infinite linear;
    > svg > circle {
      stroke-width: 4;
      stroke-linecap: round;
      fill: none;
      stroke: $grey;
      stroke-dasharray: 195;
      stroke-dashoffset: 195;
      animation: grow 3.5s infinite linear;
    }
  }

  @keyframes grow {
    0% {stroke-dashoffset: 195;}
    50% {stroke-dashoffset: 0;}
    100% {stroke-dashoffset: 195;}
  }

  @keyframes spin {
    0% {transform: translate(-50%, -50%) scale(1) rotate(0deg);}
    50% {transform: translate(-50%, -50%) scale(1) rotate(720deg);}
    100% {transform: translate(-50%, -50%) scale(1) rotate(1800deg);}
  }
}
