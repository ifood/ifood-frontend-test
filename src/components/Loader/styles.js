import styled from 'styled-components'

export const LoaderWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100%;
`
