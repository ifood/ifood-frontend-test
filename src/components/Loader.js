import styled from 'styled-components'
import { bool } from 'prop-types'
import { colors } from '../assets/styles/default-style'

const Loader = ({ centered }) => {
  return <LoaderStyle centered={centered} />
}

const LoaderStyle = styled.div`
  animation: spin 1s linear infinite;
  border: 5px solid ${colors.pink};
  border-left-color: ${colors.purple};
  border-radius: 50%;
  height: 30px;
  margin: ${({ centered }) => (centered ? '0 auto' : '0')};
  width: 30px;

  @keyframes spin {
    to {
      transform: rotate(360deg);
    }
  }
`

Loader.propTypes = {
  centered: bool
}

Loader.defaultProps = {
  centered: false
}

export default Loader
