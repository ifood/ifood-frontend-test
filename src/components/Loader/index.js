/**
 *
 * Loader
 *
 */

import React from 'react'

import LoaderGif from './loader.gif'
import { LoaderWrapper } from './styles'

function Loader() {
  return (
    <LoaderWrapper>
      <img src={LoaderGif} alt="Loading..." />
    </LoaderWrapper>
  )
}

export default Loader
