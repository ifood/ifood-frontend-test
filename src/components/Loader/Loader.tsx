import React from 'react';

import './Loader.scss';

const Loader = () => {
  return (
    <div className="loader">
      <div className="move">
        <svg viewBox="0 0 64 64">
          <circle cx="32" cy="32" r="30" />
        </svg>
      </div>
    </div>
  )
};

export default Loader;
