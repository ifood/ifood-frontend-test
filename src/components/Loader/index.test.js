import React from "react";
import { render } from "@testing-library/react";

import Loader from "./index";

test("<Loader /> renders correcly", () => {
  const component = <Loader />;
  const { asFragment } = render(component);
  expect(asFragment(component)).toMatchSnapshot();
});
