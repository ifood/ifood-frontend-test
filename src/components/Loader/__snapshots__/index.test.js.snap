// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> renders correcly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx jXCDNQ"
  />
</DocumentFragment>
`;
